src/Swarm/TUI/Controller.hs

```diff
@@ -949,6 +953,51 @@ handleREPLEventPiloting x = case x of
       & replPromptText .~ nt
       & replPromptType .~ CmdPrompt []
 
+runBaseWebCode :: MonadState AppState m => T.Text -> m ()
+runBaseWebCode uinput = do
+  s <- get
+  let topCtx = topContext s
+  unless (s ^. gameState . replWorking) $
```

Comment:
Alternatively, we could perform this `replWorking` check before we send the `RunWebCode` event over the channel, so that we could return a "rejected" status via the web API.

Or maybe we need to do the check in both places to avoid a race condition.

Reply:
That could be nice I suppose, I don't really care.  But yes, I think it would not be correct to check *only* before sending the `RunWebCode` event.  We have to check here in any case, because the base could start running in between sending and receiving/processing the event.